pkg/syncer/spec/spec_controller.go

```diff
@@ -96,7 +99,56 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, upstreamClusterName logic
 				c.AddToQueue(gvr, obj)
 			},
 		})
-		klog.InfoS("Set up informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+		klog.InfoS("Set up upstream informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+
+		downstreamInformers.ForResource(gvr).Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(obj interface{}) {
```

Comment:
Likely we want a queue for this, or we are edge-driven and transient failures leave in a broken state forever

Reply:
A queue for what, beyond the once this code is adding to?

Reply:
For deleted keys - so we can ensure they end up in the queue this is adding to? What do we want to allow in case some of the intermediate logic here fails and we can't queue? With the current setup we'd just sit around until a relist with no object on the downstream cluster, right?

Reply:
So when a downstream deployment is deleted, you want to stick the key for it into a queue?

If the syncer was offline while the downstream deployment was deleted, when it restarts, the normal upstream->downstream logic will kick in and upsert the deployment into downstream.

Reply:
Talked to Steve offline. Any error in this delete handler is unlikely (type assertions should succeed, namespaces should exist, etc.). If we were to add the key to a separate queue, we would almost certainly hit the exact same error(s) again, i.e. permanent fail.